kip-0003.md

```diff
@@ -0,0 +1,49 @@
+
+# Block Sampling
+
+We propose, instead of using all of the blocks in the past, to sample a subset of the blocks to be used for difficulty adjustment. The sampling method should be:
+ - Deterministic: it could be calculated from the DAG and would turn out the same for all nodes
+ - Uniformly distributed: all blocks have a similar chance to be sampled
+ - Incremental: it should not depend on the point of view of a particular block, so it could be inherited by future blocks
+ - Secure: a miner should not be able to cheaply affect the probability of their block to be sampled, the cost should be comparable to discarding a valid block
+
+We propose the following sampling method: to sample one in every C blocks, only consider blocks whose header hashes to a value with log(C) trailing 0s. There is no need for the hash used to be the same as the one used for proof-of-work, we propose using Blake for efficiency.
+
+(note that this method only allows choosing C to be a power of 2, there are methods to refine this. For example, by limiting the first three bits to be 0, and the last two bits to contain *at least one* 0, we get a probability of (1/8)\*(3/4) = 3/32 for each block to be sampled. I see no reason to treat this problem generally, whatever sampling rate we choose, we could tailor a sampling rule ad-hoc that is sufficiently close)
+
+That this method is deterministic and incremental is true-by-design, the uniformity follows from assuming the standard assumption that black sufficiently approximates a random oracle. The security follows from the fact that the nonce is included in the header, so (assuming Blake and kHH distribute independently), the expected number of hashes required to find a nonce that makes the block both valid and selected is C times larger than the hashes required for a valid block. It could be argued that a negligible factor of (C-1)/C is required for the miner to adversarialy mine a block which is not selected (whereby harming the sampling process), however, the memorylessness of the Poissin process implies that, conditioned on the fact that the miner found a nonce for a sampled block, they still have the same expected waiting time as working on a new block.
+
+# Regulating Emissions
+
+We currently regulate emissions by not rewarding blocks outside the difficulty window. This approach is unsuitable when the difficulty window is sampled. So instead, we only reward blocks whose accumulated blue score is at least as much as the lowest accumulated blue score witnessed in the difficulty window.
+
+# Proposed Constants
+
+I propose the following:
+ - Increase the allowed expected drift from two minutes to 10 minutes. This requires a window time of 20 minutes. I propose a sample rate of a block per 10 seconds. The overall size of the timestamp flexibility window would then be 121 blocks.
```

Comment:
same comment about the item list, please check the preview to make sure it compiles visually the way you want it